src/librustc_typeck/diagnostics.rs

```diff
@@ -380,6 +380,52 @@ fn main() {
 ```
 "##,
 
+E0044: r##"
+You can't use type parameters on foreign items. Example of erroneous code:
+
+```
+extern { fn some_func<T>(); }
+```
+
+Just remove the type parameter to make this code works:
```

Comment:
You should also have more advice here. "Removing the type parameter" won't make this code work, because now you need concrete types here.

Something along the lines of manually putting in the versions that you need (manual monomorphization) is what we want here